optimade/models/structures.py

```diff
@@ -893,6 +895,29 @@ def check_anonymous_formula(cls, v):
 
         return v
 
+    @validator("chemical_formula_anonymous", "chemical_formula_reduced")
+    def check_reduced_formulae(cls, value, field):
+        if value is None:
+            return value
+
+        numbers = [n.strip() or 1 for n in re.split(r"[A-Z][a-z]*", value)]
+        # Need to remove leading 1 from split and convert to ints
+        numbers = [int(n) for n in numbers[1:]]
+
+        if any(n == 0 for n in numbers):
```

Comment:
Could you not simply write 
```suggestion
        if 0 in numbers:
```
Or if you want to be thorough
```suggestion
        if any(n <= 0 for n in numbers):
```

Reply:
Good point, I just followed the older validator we had for the anonymous formula, I can change it there too.